### PR TITLE
Move to 24.04 for docker tests

### LINKED ISSUE #3147: `-test.kill-daemon` does not work on Ubuntu 24.04 (`Failed to kill unit containerd.service: Failed to send signal SIGKILL to auxiliary processes: Invalid argument\n`)
https://github.com/containerd/nerdctl/blob/a5445322e9ed39ce18d82c8391fdbc7a84545855/pkg/testutil/testutil.go#L564-L573

### PR DESCRIPTION
**Pull request by apostasie**
I am assuming this will fail until we fix the issue in #3147 but curious what else is there to fix.

Draft for now.

### CODE REVIEW

**Comment by apostasie** [comment]
Rebasing on top of #3177
Let's see what goes.

**Comment by apostasie** [comment]
List of amendments:

* fixing `--kill.daemon` (other PR)

* TestRunCgroupParent - invalid cgroup name to compare with (faulty assumption that nerdctl was the only one of the two with systemd cgroup)

* TestTailFollowRotateLogs - somewhat unrelated, but a significant source of flakyness it seems - using Combined to test is a mistake, too short a timeout, not checking for empty lines - also, our code needs to be fixed as well (applied a bandaid in the meantime)

File: pkg/logging/json_logger.go [hunk 1]
```diff
@@ -29,6 +29,7 @@ import (
 	"time"
 
 	"github.com/containerd/containerd/runtime/v2/logging"
+	"github.com/containerd/errdefs"
```
Comment (AkihiroSuda) [inline pkg/logging/json_logger.go:32]:
```suggestion
	"github.com/containerd/containerd/errdefs"
```
(Should be changed to github.com/containerd/errdefs later in another PR)

Comment (apostasie) [reply pkg/logging/json_logger.go:32]: Nice catch, thanks.

**Comment by apostasie** [comment]
Rebasing against main for good measure.

**Comment by apostasie** [comment]
1 failure - IPFS

I think this is good to go @AkihiroSuda

File: pkg/logging/json_logger.go [hunk 2]
```diff
@@ -129,7 +130,15 @@ func (jsonLogger *JSONLogger) PostProcess() error {
 func viewLogsJSONFile(lvopts LogViewOptions, stdout, stderr io.Writer, stopChannel chan os.Signal) error {
```
Comment (AkihiroSuda) [inline pkg/logging/json_logger.go:130]: Is this function called in docker tests?

Comment (apostasie) [reply pkg/logging/json_logger.go:130]: Indirectly, yes

Comment (apostasie) [reply pkg/logging/json_logger.go:130]: Do you mean "is it called with target=docker"?

Comment (AkihiroSuda) [reply pkg/logging/json_logger.go:130]: Right, probably this should have another GitHub issue and PR?

Comment (apostasie) [reply pkg/logging/json_logger.go:130]: Oh, I see what you mean. Yes, I guess I can split it out and submit a separate PR.

Comment (apostasie) [reply pkg/logging/json_logger.go]: Pushed again without the above (will submit as separate PR).

**Comment by apostasie** [comment]
@AkihiroSuda done

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**